src/components/post-card/PostCard.js

```diff
@@ -8,57 +8,83 @@ import './PostCard.scss';
+    this.state.showComments === false ?
+      this.setState({selectedLabel:hide}) :
+      this.setState({selectedLabel:show})
   };
 
-  return (
-    <div className="may-post-card card">
-      {hasImage && (
-        <div className="may-post-card-img" id="my-block" onClick={() => {
-          alert('ghvcdhfvbdfsjvbdf');
-        }}>
-          <img src={kittyUrl} />
-        </div>
-      )
-      }
-      {!hasImage && (
-        <div className="may-post-card-img">
-          <img src={DefaultImg} />
-        </div>
-      )}
+  render() {
+      const { post, hasImage, comments = [], author } = this.props;
+      // console.log('hasImage', hasImage);
+      const { title, body } = post;
+      const kittyUrl = `https://cataas.com/cat/says/hello%20world!?${Math.random() * 1000}`;
+      const { showComments, selectedLabel } = this.state;
+
+      const renderImage = () => {
+        return hasImage ? (
+            <img src={kittyUrl} />
+        ) : (
+            <img src={DefaultImg} />
+        );
+      };
 
-      <div className="card-body">
-        <h4 className="card-title title">{title}</h4>
-        <div className="card-text body">
-          {body}
-        </div>
-        <blockquote className="blockquote">
-          <footer className="blockquote-footer">Author:
-            <cite title="Source Title">{author}</cite>
-          </footer>
-        </blockquote>
-      </div>
+      return (
+          <div className="may-post-card card">
+            { hasImage && (
+                <div className="may-post-card-img" id="my-block" onClick={() => {
+                  alert("its fun, isn't it?");
+                }}>
+                  <img src={ kittyUrl } />
+                </div>
+            )
+            }
+            { !hasImage && (
+                <div className="may-post-card-img">
+                  <img src={ DefaultImg } />
+                </div>
+            )}
 
-      {!!comments.length && <label>Comments:</label> }
-      {
-        comments.map(comment => (<Comment comment={comment} key={comment.id}/>))
-      }
-    </div>
-  );
+            <div className="card-body">
+              <h4 className="card-title title">{ title }</h4>
+              <div className="card-text body">
+                {body}
+              </div>
+              <blockquote className="blockquote">
+                <footer className="blockquote-footer">Author:
+                  <cite title="Source Title">{ author }</cite>
+                </footer>
+              </blockquote>
+            </div>
+
+            { !!comments.length && <button className={ showComments === false ?
+                                                                      'btn btn-success' :
+                                                                      "btn btn-outline-danger"}
+                                           onClick={this.changeButton}>
+                                    { selectedLabel }
```

Comment:
Here instead of using `selectedLabel ` from state you can just use ternary condition

{ showComments ? 'hide comments' : 'show comments' }

and this allows you to get rid of selectedLabel in state completely

Reply:
But you variant is ok too